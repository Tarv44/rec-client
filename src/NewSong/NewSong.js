import { Component } from 'react';
import './NewSong.css'

export default class NewSong extends Component {
    render() {
        return (
            <>
                <div className='exchange-form-group'>
                    <label htmlFor={`new-song-title-${this.props.index}`}>Title</label>
                    <input 
                        type='text' 
                        name='song-title' 
                        id={`new-song-title-${this.props.index}`}
                        onChange={e => this.props.handleTitle(e.target.value, this.props.index)}
                    />
                </div>
                <div className='exchange-form-group'>
                    <label htmlFor={`new-song-link-${this.props.index}`}>Link</label>
                    <input 
                        type='text' 
                        name='song-link' 
                        id={`new-song-link-${this.props.index}`}
                        onChange={e => this.props.handleLink(e.target.value, this.props.index)}
                    />
                </div>
                <div className='exchange-form-group'>
                    <label htmlFor={`new-artist-${this.props.index}`}>Artist</label>
                    <input 
                        type='text' 
                        name='artist' 
                        id={`new-artist-${this.props.index}`}
                        onChange={e => this.props.handleArtist(e.target.value, this.props.index)}
                    />
                </div>
                <div className='exchange-form-group'>
                    <label htmlFor={`new-album-${this.props.index}`}>Album</label>
                    <input 
                        type='text' 
                        name='album' 
                        id={`new-album-${this.props.index}`}
                        onChange={e => this.props.handleAlbum(e.target.value, this.props.index)}
                    />
                </div>
            </>
        )
    }
}