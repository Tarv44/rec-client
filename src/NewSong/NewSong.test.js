import React from 'react'
import ReactDom from 'react-dom'
import renderer from 'react-test-renderer'
import NewSong from './NewSong'
import { BrowserRouter } from 'react-router-dom'


describe('----- NewSong -----', () => {
    const form_state = {
        title: '',
        url_link: '',
        artist: '',
        album: ''
    }

    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDom.render((
            <BrowserRouter>
                <NewSong index={1} form_state={form_state}/>
            </BrowserRouter> 
        ), div)
        ReactDom.unmountComponentAtNode(div);
    })
    it('renders the UI as expected', () => {
        const comp = renderer
            .create(
                <BrowserRouter>
                    <NewSong index={1} form_state={form_state}/>
                </BrowserRouter> 
            )
            .toJSON()
        expect(comp).toMatchSnapshot()
    })
})