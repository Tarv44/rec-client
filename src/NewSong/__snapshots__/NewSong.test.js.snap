// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`----- NewSong ----- renders the UI as expected 1`] = `
Array [
  <div
    className="exchange-form-group"
  >
    <label
      htmlFor="new-song-title-1"
    >
      Title
    </label>
    <input
      id="new-song-title-1"
      name="song-title"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>,
  <div
    className="exchange-form-group"
  >
    <label
      htmlFor="new-song-link-1"
    >
      Link
    </label>
    <input
      id="new-song-link-1"
      name="song-link"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>,
  <div
    className="exchange-form-group"
  >
    <label
      htmlFor="new-artist-1"
    >
      Artist
    </label>
    <input
      id="new-artist-1"
      name="artist"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>,
  <div
    className="exchange-form-group"
  >
    <label
      htmlFor="new-album-1"
    >
      Album
    </label>
    <input
      id="new-album-1"
      name="album"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>,
]
`;
